yext/answers-search-ui
# support custom prompts in autocomplete

this pr provide support for user to add custom autocomplete prompts in addition to the autocomplete results from core. 

User can either specify an array of string for custom prompts through searchbar.autocomplete config (`ANSWERS.addComponent('SearchBar', {..., autocomplete: {customPrompts: ['prompt']} }`) or use `setCustomPrompts` (use in theme). In autocomplete's setState, custom prompts will transform into right data format and merge with core results. This is then set in storage to trigger a new state update with the combined result. Custom prompts appear on empty search and persist when user type similar substring as the prompt.

J=SLAP-1803
TEST=manual

- serve page and add SearchBar with customPrompts configuration, with and without hardcoded prompts from backend. See that custom prompts appear as expected. 
  - Type same starting substrings and see that custom prompts still display (levenshtein match) 
  - Type and remove input, see that custom prompts appear again.
- Hook SDK changes to theme. spin up test site and see that custom prompts appear as expected
